### PR TITLE
refactor: modularize docker compose and migration scripts

### PR DESCRIPTION
**Pull request by Lenbkr**
## What did we change?
Modularize docker compose yaml 
Add Auth/Keycloak

## How was it tested?
- [x] Locally
- [ ] Development Environment
- [ ] Not needed, changes very basic
- [ ] Husky successfully run
- [ ] Unit tests passing and Documentation done

### CODE REVIEW

**Review by JeanPierreNell — APPROVED** [review]

**Review by rtkay123 — APPROVED** [review]

File: .env [hunk 1]
```diff
@@ -12,10 +12,12 @@ TADP_BRANCH=main
...
 CMS_BRANCH=main
 BATCH_PPA_BRANCH=main
-ADMIN_BRANCH=dev
+ADMIN_BRANCH=main
+AUTH_SERVICE_BRANCH=main
+EVENT_FLOW_BRANCH=main
 
 # Ports
-TMS_PORT=5000
+TMS_PORT=3000
```
Comment (Justus-at-Tazama) [inline .env:20]: Why are we moving this back to 3000?

File: README.md [hunk 2]
```diff
@@ -81,34 +78,48 @@ GH_TOKEN=${GH_TOKEN}
 # Branches
 TMS_BRANCH=main
 ED_BRANCH=main
+RULE_EXECUTER_BRANCH=main
```
Comment (Justus-at-Tazama) [inline README.md:81]: OK, so sure, the thing we are compiling is the rule executor, but the result is the rule _processor_ and it helps if this is unique if we are also then adding the other rule processors. Why are we rolling back this change?

File: README.md [hunk 3]
```diff
@@ -81,34 +78,48 @@ GH_TOKEN=${GH_TOKEN}
...
+NATS_UTILITIES_BRANCH=main
+CMS_BRANCH=main
+BATCH_PPA_BRANCH=main
+ADMIN_BRANCH=main
+AUTH_SERVICE_BRANCH=main
+EVENT_FLOW_BRANCH=main
 
 # Ports
-TMS_PORT=5000
+TMS_PORT=3000
```
Comment (Justus-at-Tazama) [inline README.md:92]: Ditto - the current postman testing environments for the docker deployment all point to port 5000...

File: README.md [hunk 4]
```diff
@@ -117,40 +128,23 @@ You'll be able to access the web interfaces for the deployed components through
...
 
-```
-docker compose up -d tms ed tp tadp
-```
-
-
-This command will install:
-
- - The Transaction Monitoring Service API at `<https://localhost:5000>`, where messages will be sent for evaluation. 
+ - The Transaction Monitoring Service API at `<https://localhost:5000>`, where messages will be sent for evaluation. (Port configured in .env file under `TMS_PORT`)
```
Comment (Justus-at-Tazama) [inline README.md:143]: Case in point...

File: README.md [hunk 5]
```diff
@@ -224,6 +229,10 @@ List of \<services\>
 - tadp  
 - tp  
 - rule-901
```
Comment (Justus-at-Tazama) [inline README.md:231]: This is also why I'd prefer the rule-executer branch to reflect the rule processor id and not the rule-executer; it's a bit confusing having the .env branch variable say one thing, and then the docker-compose.yaml saying another...

**Review by Justus-at-Tazama — CHANGES_REQUESTED** [review]

**Review by rtkay123 — DISMISSED** [review]

**Review by rtkay123 — APPROVED** [review]

**Review by Justus-at-Tazama — APPROVED** [review]

**Pull request merged**